ra2ce/graph/hazard.py

```diff
@@ -516,6 +517,42 @@ def join_table(
         )  # Check if this is the right name
         return graph
 
+    def get_point_hazard_from_network(
+        self, points: gpd.GeoDataFrame, network: gpd.GeoDataFrame
+    ) -> gpd.GeoDataFrame:
+        """get hazard informations for points at nearest network locations.
+
+        Args:
+            points (gpd.GeoDataFrame): points of interest, will be snapped to the network by nearest distance method.
+            network (gpd.GeoDataFrame): network of interest
+
+        Returns:
+            gpd.GeoDataFrame: points with relation to network and hazard informations
+
+        Raise:
+            error: if node id columns do not exisit in the network
+        """
+
+        if "node_A" in network.columns:  # shapefiles
+            network["edge_fid"] = [
+                f"{na}_{nb}" for na, nb in network[["node_A", "node_B"]].values
+            ]
+        elif "u" in network.columns:  # osm
+            network["edge_fid"] = [
+                f"{na}_{nb}" for na, nb in network[["u", "v"]].values
+            ]
+        else:
+            logging.error(
+                "Node id columns are not found in the network. Support node_A, node_B or u, v"
+            )
+
+        points = gpd.sjoin_nearest(
+            points, network, how="left", distance_col="edges_distance"
+        )
+        if points["edges_distance"].isna().sum() > 0:
```

Comment:
I believe this is better with the 'any' operator.
```suggestion
        if any(points["edges_distance"].isna()):
```